test/fail_compilation/fail15089.d

```diff
@@ -0,0 +1,10 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail15089.d(10): Error: cannot implicitly convert expression (130) of type int to byte
```

Comment:
I see that the error message now points to the correct line. But unfortunately it still prints an expression that's not in the source code, `130`. What happens if I do something stupid like putting two expressions on the same line that result in the same value:

```d
byte piece = Pieces.Rook ^ color; byte piece2 = 129 + 1;
```

Ideally it should print both the original expression from the source code and that value it was evaluated to.